Bump Go and Dockerfile versions

macedogm opened the pull request:
This PR updates:

1. Move `Dockerfile.dapper` to use BCI Golang `bci/golang:1.19`.
2. Move the main CLI image in `package/Dockerfile` to use BCI Base `bci/bci-base:15.5`.
3. Move the main CLI image in `package/Dockerfile` to run as a non-root user.
4. Bump Go to `1.19` in `go.mod`.
5. Several bumps in `go.mod` to fix CVEs:
   1. `github.com/rancher/wrangler`
   2. `golang.org/x/crypto`
   3. `golang.org/x/net`
   4. `golang.org/x/sys`
   5. `golang.org/x/term`
   7. `golang.org/x/text`
   8. No other major dependency was bumped besides the ones listed.
6. Bump `rancher/dapper` to `v0.6.0` in `.drone.yml`.
7. Run `go fmt` on `config/config.go` to fix a small formatting issue.

pjbgf commented:
The symptom of `-race` no longer working could be related to the fact that something on the changes broke CGO - which `-race` relies on.

This project does not use `CGO` directly, however some of its dependencies does (e.g. docker). Therefore, this may result in specific execution paths having a different behaviour (or may panic at runtime) due to the lack of CGO - i.e. some specific features may not be supported.

superseb commented:
What does `it was not working properly.` mean? I have removed it from RKE https://github.com/rancher/rke/pull/3174 because it needs an additional package to be installed to be used.

```
  | go1.19-race                      | Go runtime race detector                                               | package
```

macedogm commented:
@superseb you are right, I somehow missed that `bci/golang` doesn't have the `go1.19-race` package. Locally it worked for me, because it's available in Tumbleweed. I updated the PR to re-add the `-race` flag to `go test`.

Will check offline with the BCI team if they can add the package.

macedogm commented:
FYI the BCI team said that they will add `go1.xx-race` package in the next Go BCI image update.

cmurphy commented on .drone.yml:
```diff
@@ -9,7 +9,7 @@ platform:
 steps:
 - name: build
   pull: default
-  image: rancher/dapper:1.10.3
```
Why is this being downgraded?

pjbgf replied:
I think the versioning changed overtime, the latest is `v0.6.0`:

![image](https://github.com/rancher/cli/assets/5452977/a786f737-5180-4153-a620-2714db6a1fcf)
![image](https://github.com/rancher/cli/assets/5452977/b05da5ea-aebc-4fb2-a30b-149c0949bddb)

This aligns with that repository Git tags.

macedogm replied:
This ^ is right. [v0.6.0](https://github.com/rancher/dapper/releases/tag/v0.6.0) is from Jul 7, 2022 and 1.10.3 is not even in current [Dapper's repo](https://github.com/rancher/dapper/releases?page=2).

JonCrowther commented on Dockerfile.dapper:
```diff
@@ -1,21 +1,16 @@
+RUN GO_VERSION=$(echo "$GOLANG_VERSION" | sed -r 's/^([[:digit:]]+.[[:digit:]]+).*/\1/') && \
+    zypper -n install docker go${GO_VERSION}-race
 
+ENV GOLANGCI_LINT v1.53.3
```
Will renovate bot be able to identify this is a version as an ENV? In Webhook it puts the version on the `curl` command and renovate bot is able to update this automatically:
https://github.com/rancher/webhook/pull/251/files

pjbgf replied:
[Not at the moment](https://github.com/rancher/renovate-config/blob/5d08c09fac6c3f2d27474fb6996bc6a0766776d9/default.json#L83-L93), but we probably should update the renovate config to accommodate that.

macedogm commented:
> @superseb you are right, I somehow missed that `bci/golang` doesn't have the `go1.19-race` package. Thanks for noticing it. Locally it worked for me, because it's available in Tumbleweed. I updated the PR to re-add the `-race` flag to `go test`.
> 
> Will check offline with the BCI team if they can add the package.

Removed the hacky code to add go-racer package, because it's now inside the Go base image by default:

```shell
> podman run --rm -it registry.suse.com/bci/golang:1.19 rpm -aq | grep go
go1.19-doc-1.19.11-150000.1.37.1.x86_64
libgomp1-12.3.0+git1204-150000.1.10.1.x86_64
go1.19-1.19.11-150000.1.37.1.x86_64
go1.19-race-1.19.11-150000.1.37.1.x86_64
```

CC @superseb and @pjbgf
